src/Airy/cairy.jl

```diff
@@ -377,197 +377,98 @@ end
+airy_large_argument_cutoff(z::ComplexOrReal{Float64}) = abs(z) > 8.3
 airy_large_argument_cutoff(z::ComplexOrReal{Float32}) = abs(z) > 4
 
-function airyai_large_argument(x::Real)
-    x < zero(x) && return real(airyai_large_argument(complex(x)))
-    return airy_large_arg_a(abs(x))
-end
-
-function airyai_large_argument(z::Complex{T}) where T
-    x, y = real(z), imag(z)
-    a = airy_large_arg_a(z)
-    if x < zero(T) && abs(y) < 5.5
-        b = airy_large_arg_b(z)
-        y >= zero(T) ? (return a + im*b) : (return a - im*b)
-    end
-    return a
-end
-
-function airyaiprime_large_argument(x::Real)
-    x < zero(x) && return real(airyaiprime_large_argument(complex(x)))
-    return airy_large_arg_c(abs(x))
-end
-
-function airyaiprime_large_argument(z::Complex{T}) where T
-    x, y = real(z), imag(z)
-    c = airy_large_arg_c(z)
-    if x < zero(T) && abs(y) < 5.5
-        d = airy_large_arg_d(z)
-        y >= zero(T) ? (return c + im*d) : (return c - im*d)
-    end
-    return c
-end
-
-function airybi_large_argument(x::Real)
-    if x < zero(x)
-        return 2*real(airy_large_arg_b(complex(x)))
+function airyai_large_args(z::Complex{T}) where T
+    if imag(z) < zero(T)
+        out = conj.(airyaix_large_args(conj(z)))
     else
-        return 2*(airy_large_arg_b(x))
+        out = airyaix_large_args(z)
     end
+    return out .* exp(-2/3 * z * sqrt(z))
 end
-
-function airybi_large_argument(z::Complex{T}) where T
-    x, y = real(z), imag(z)
-    b = airy_large_arg_b(z)
-    abs(y) <= 1.7*(x - 6) && return 2*b
-
-    check_conj = false
-    if y < zero(T)
-        z = conj(z)
-        b = conj(b)
-        y = abs(y)
-        check_conj = true
-    end
-
-    a = airy_large_arg_a(z)
-    if x < zero(T) && y < 5
-        out = b + im*a
-        check_conj && (out = conj(out))
-        return out
-    else
-        out = 2*b + im*a
-        check_conj && (out = conj(out))
-        return out
-    end
-end
-
-function airybiprime_large_argument(x::Real)
-    if x < zero(x)
-        return 2*real(airy_large_arg_d(complex(x)))
+function airybi_large_args(z::Complex{T}) where T
+    if imag(z) < zero(T)
+        out = conj.(airybix_large_args(conj(z)))
     else
-        return 2*(airy_large_arg_d(x))
+        out = airybix_large_args(z)
     end
+    return out .* exp(2/3 * z * sqrt(z))
 end
 
-function airybiprime_large_argument(z::Complex{T}) where T
-    x, y = real(z), imag(z)
-    d = airy_large_arg_d(z)
-    abs(y) <= 1.7*(x - 6) && return 2*d
-
-    check_conj = false
-    if y < zero(T)
-        z = conj(z)
-        d = conj(d)
-        y = abs(y)
-        check_conj = true
-    end
-
-    c = airy_large_arg_c(z)
-    if x < zero(T) && y < 5
-        out = d + im*c
-        check_conj && (out = conj(out))
-        return out
+@inline function airyaix_large_args(z::Complex{T}) where T
+    xsqr = sqrt(z)
+    xsqrx =  Base.FastMath.inv_fast(z * xsqr)
+    A, B, C, D = compute_airy_asy_coef(z, xsqrx)
+    
+    if abs(angle(z)) > 4π/6
```

Comment:
rewrite in terms of ratio of components. `angle` is slow. (i.e. something like `if abs(real(z)) < sqrt(3)*abs(imag(z))`)

Reply:
There might be a better way to do it but it did shave off ~10ns